lib/postgres.go

```diff
@@ -0,0 +1,1414 @@
+package lib
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/dgraph-io/badger/v3"
+	"github.com/go-pg/pg/v10"
+	"github.com/go-pg/pg/v10/orm"
+	"github.com/golang/glog"
+)
+
+type Postgres struct {
+	db *pg.DB
+}
+
+func NewPostgres(db *pg.DB) *Postgres {
+	// Print all queries.
+	//db.AddQueryHook(pgdebug.DebugHook{
+	//	Verbose: true,
+	//})
+
+	return &Postgres{
+		db: db,
+	}
+}
+
+func LogSelect(query *orm.Query) error {
+	selectQuery := orm.NewSelectQuery(query)
+	fmter := orm.NewFormatter().WithModel(selectQuery)
+	queryStr, _ := selectQuery.AppendQuery(fmter, nil)
+	glog.Info(string(queryStr))
+	return query.Select()
+}
+
+//
+// Tables
+//
+// When we can, we use unique fields (or combinations of unique fields) as the primary keys on the models.
+// This lets us use the WherePK() query while also minimizing columns and indicies on disk.
+//
+
+type PGChain struct {
+	tableName struct{} `pg:"pg_chains"`
+
+	Name    string `pg:",pk"`
```

Comment:
Just being lazy: Why do the columns have commas?

Reply:
I don't really get the annotation scheme here. Maybe add a comment under the Tables explaining it and give examples.